frontend/src/cljs/imas_seamap/map/events.cljs

```diff
@@ -652,13 +652,40 @@
 (defmulti get-layer-legend-success #(get-in %2 [1 :layer_type]))
 
 (defmethod get-layer-legend-success :wms
-  [db [_ {:keys [id] :as _layer} response]]
-  (let [legend (->> response :Legend first :rules
-                    (mapv
-                     (fn [{:keys [title filter symbolizers]}]
-                       {:label  title
-                        :filter filter
-                        :color  (-> symbolizers first :Polygon :fill)})))]
+  [db [_ {:keys [id server_url layer_name] :as _layer} response]]
+  (let [legend (cond
+                 (-> response :Legend first :rules first :symbolizers first :Raster)
+                 (append-query-params-from-map
+                  server_url
+                  {:REQUEST     "GetLegendGraphic"
+                   :LAYER       layer_name
+                   :TRANSPARENT true
+                   :SERVICE     "WMS"
+                   :VERSION     "1.1.1"
+                   :FORMAT      "image/png"})
+
+                 :else
+                 (->> response :Legend first :rules
+                      (mapv
+                       (fn [{:keys [title filter symbolizers]}]
+                         {:label  title
+                          :filter filter
+                          :style
+                          (cond
+                            (-> symbolizers first :Polygon)
+                            {:background-color (-> symbolizers first :Polygon :fill)
+                             :height           "100%"
+                             :width            "100%"}
+
+                            (-> symbolizers first :Point)
+                            (let [{:keys [graphics size]} (-> symbolizers first :Point)
+                                  {:keys [mark fill stroke stroke-width]} (first graphics)]
+                              (merge
+                               {:background-color fill
+                                :border           (str "solid " stroke-width "px " stroke)
+                                :width            (str size "px")
+                                :height           (str size "px")}
+                               (when (= mark "circle") {:border-radius "100%"}))))}))))]
```

Comment:
Just from he size of this diff I'm in favour of extracting that out into a separate function.  (not sure after that if it's any further benefit making it a `defmulti` or just extracting the code as-is; see what you think)

Reply:
Done